### PR TITLE
documentation

### LINKED ISSUE #387: Include SHA hash of downloadables
We should be hashing anything that we have stored on GCS that doesn't get checked into PyPI.  Right now, it seems like that would only include the Int64 builds of TF we use for OS X & Linux.

### PR DESCRIPTION
**Pull request by mortendahl**
- [x] more detailed installation instructions
- [x] guidelines for merging
- [x] small refresh of gcp debian install script
- [x] instructions for `make test`
- [x] macos installation instructions
- [x] fingerprints for downloads

Closes #386
Closes #387

### CODE REVIEW

**Comment by mortendahl** [comment]
missing: instructions for getting `make test` to work!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,122 +1,60 @@
...
+To contribute, [fork](https://help.github.com/articles/fork-a-repo/) `tf-encrypted`, commit your changes, and [open a pull request](https://help.github.com/articles/using-pull-requests/).
 
-**Begin Developing**
+While you may be asked to make changes to your submission during the review process, we will work with you on this and suggest changes. Consider giving us [push rights to your branch](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) so we can potentially also help via commits.
 
-You should now be able to begin developing! You can check that your setup works by running our tests via `make test`.
+## Commit history and merging
 
-### Finding Your First Issue
+For the sake of transparency our key rule is to keep a logical andn intelligible commit history, meaning anyone stepping through the commits on either the `master` branch or as part of a review should be able to easily follow the changes made and their potential implications.
```
Comment (justin1121) [inline .github/CONTRIBUTING.md]: extra character andn vs and

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,122 +1,60 @@
...
+Some guidelines:
 
-If you ever need help, don't hesitate to ask!
+- even simple code changes such as moving code around can obscure semantic changes, and in those case there should be two commits: one that e.g. only moves code (with a note of this in the commit description) and one that performs the semantic change
 
-# Reporting a bug
+- progressions that have no logical justification for being split into several commits should be squeezed
 
-Think you've found a bug? Let us know by opening an [issue in our tracker](https://github.com/mortendahl/tf-encrypted/issues) and apply the "bug" label!
+- code does not have to compile or pass all tests at each commit; but leave a remark and plan in the commit description so a reviewer is aware and can plan accordingly
```
Comment (justin1121) [inline .github/CONTRIBUTING.md]: punctuation?

File: INSTALL.md [hunk 3]
```diff
@@ -39,23 +39,45 @@ That's it - you should now have a working copy ready for development!
...
-``` 
+./tf-encrypted/ $ docker pull tensorflow/tensorflow:custom-op
+```
+
+Note: Docker commands may have to be run with sudo!
+
+Once the docker image has been retrieved we need to open up a shell to input the following commands.
+
+```
+./tf-encrypted/ $ docker run -it -v `pwd`:/opt/tf_encrypted \
```
Comment (mortendahl) [inline INSTALL.md]: this is only on linux right? ie no need to go through this on macOS?

File: INSTALL.md [hunk 4]
```diff
@@ -106,10 +128,10 @@ then create a virtual environment (on Ubuntu 16.04 we needed to `export LC_ALL=C
 
 and follow the [basic instructions](#basics).
 
-To compile custom ops first install the additional system tools
+To compile custom ops first install docker
 
 ```
-$ sudo apt install -y libtool automake g++ curl
+./ $ apt install docker.io
```
Comment (mortendahl) [inline INSTALL.md]: could it be `sudo apt install -y docker.io` to fit with the others?

File: INSTALL.md [hunk 5]
```diff
@@ -39,23 +39,45 @@ That's it - you should now have a working copy ready for development!
...
-To do so you need the following system tools installed: curl, libtool, automake, and g++. These can typically be installed using your system's package manager (see below) and once done allows for building the custom ops using:
+Tensorflow custom ops must be built in a docker container to avoid ABI compatibility issues. First, you'll have to pull down the docker image:
 
 ```
-(venv) ./tf-encrypted/ $ make build
-``` 
+./tf-encrypted/ $ docker pull tensorflow/tensorflow:custom-op
+```
+
+Note: Docker commands may have to be run with sudo!
```
Comment (mortendahl) [inline INSTALL.md]: i had to do this; would it be bad to do it by default?

File: INSTALL.md (hunk 3 above)
Comment (justin1121) [reply INSTALL.md]: Yep don't think this will work with macos. Will update when I add the macos instructions.

File: INSTALL.md (hunk 4 above)
Comment (justin1121) [reply INSTALL.md]: Yep

File: INSTALL.md (hunk 5 above)
Comment (justin1121) [reply INSTALL.md]: Might as well!

**Pull request merged**